pallets/schemas/src/migration.rs

```diff
@@ -85,27 +67,44 @@ pub mod v1 {
 // ==============================================
 /// Schema migration to v1 for pallet-stateful-storage
 /// This struct derives OnRuntimeUpgrade trait which is used to run the migration (check runtime)
-pub struct SchemaMigration<T: Config>(PhantomData<T>);
-
-impl<T: Config> OnRuntimeUpgrade for SchemaMigration<T> {
-	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
-		let weight = v1::pre_migrate_schemas_to_v1::<T>();
-		log::info!("pre_upgrade weight: {:?}", weight);
-		Ok(Vec::new())
-	}
+pub struct Migration<T: Config>(PhantomData<T>);
 
+impl<T: Config> OnRuntimeUpgrade for Migration<T> {
 	fn on_runtime_upgrade() -> Weight {
-		v1::migrate_schemas_to_v1::<T>()
-	}
+		let version = StorageVersion::get::<Pallet<T>>();
+		let mut weight: Weight = Weight::zero();
+
+		if version < 1 {
+			weight = weight.saturating_add(v1::migrate::<T>());
```

Comment:
Is this migration not idempotent?

Reply:
This is more standardization and following a pattern. As far as I know this is the standard setup so that there are
the minimal on init code run.